hypervisor/virtualbox/virtualbox.go

```diff
@@ -114,6 +129,38 @@ func (*virtualbox) Modify(vmName string, cpus int, mem int) error {
 	return nil
 }
 
+func (*virtualbox) GetVMProperty(vmName, property string) (*string, error) {
+	prop, ok := propertyMap[property]
+	if !ok {
+		return nil, fmt.Errorf("property %s not found", property)
+	}
+
+	guestProperty, err := vboxManage("guestproperty",
+		"enumerate",
+		vmName).Call()
+
+	if err != nil {
+		return nil, err
+	}
+
+	time.Sleep(30 * time.Second)
```

Comment:
putting sleep as this get property works only when the vm is up and running, this wait time is to ensure it is up, will change this to health check based approach instead of sleep

Reply:
You can open an issue and send PR later if you like

Reply:
BTW, the health check solution needs to have some retry mechanism (similar to the one in k8s or so)

Reply:
yep i just found out that the enumerate command also returns the state of the vm i'm going to put a loop on that untill it shows up and then connect